encoding/descaling.md

```diff
@@ -1,477 +1,332 @@
-so it's worth keeping in mind.
-
-Here's an example of the previous frame
-when descaled using various kernels and settings
-(note that descale requires either GrayS, RGBS, or YUV444PS.
-I'll be using `split` and `join` from `kagefunc` to split the planes
-and then join them again in this example,
-and `get_w` from `vsutil` to calculate the width)[^1]:
-
-[Comparison between frames][manaria_compare]
-
-```Py
-from vapoursynth import core
+```py
 import vsutil
 import kagefunc as kgf
-import fvsfunc as fvf
-
-src = core.lsmas.LWLibavSource("BDMV/[BDMV][190302][マナリアフレンズ I]/BD/BDMV/STREAM/00007.m2ts")
-src = fvf.Depth(src, 32)
-
-y, u, v = kgf.split(src)
-height = 878
-width = vsutil.get_w(height)
-
-# Bilinear
-descale_a = core.descale.Debilinear(y, width, height).resize.Bilinear(1920, 1080)
-descale_a = kgf.join([descale_a, u, v])
-# Mitchell-Netravali
-descale_b = core.descale.Debicubic(y, width, height, b=1/3, c=1/3).resize.Bicubic(1920, 1080, filter_param_a=1/3, filter_param_b=1/3)
-descale_b = kgf.join([descale_b, u, v])
-# Sharp Bicubic
-descale_c = core.descale.Debicubic(y, width, height, b=0, c=1).resize.Bicubic(1920, 1080, filter_param_a=0, filter_param_b=1)
-descale_c = kgf.join([descale_c, u, v])
-# B-Spline
-descale_d = core.descale.Debicubic(y, width, height, b=1, c=0).resize.Bicubic(1920, 1080, filter_param_a=1, filter_param_b=0)
-descale_d = kgf.join([descale_d, u, v])
-# Catmull-rom
-descale_e = core.descale.Debicubic(y, width, height, b=0, c=1/2).resize.Bicubic(1920, 1080, filter_param_a=0, filter_param_b=1/2)
-descale_e = kgf.join([descale_e, u, v])
-# Spline36
-descale_f = core.descale.Despline36(y, width, height).resize.Spline36(1920, 1080)
-descale_f = kgf.join([descale_f, u, v])
-```
-
-You might realize that after descaling,
-we are immediately upscaling the frame
-with the same kernel and values again.
-This is done so we can compare the before and after.
-The closer the new frame is to the old one,
-the more likely it is that you've got the correct kernel.
-Zooming in on the frame at 4x magnification or higher
-using Nearest Neighbor
-will help immensely.
-An alternative that you can use
-is to simply descale until
-you've got what you believe to be the best result.
-It's faster to do it this way,
-but might be less accurate.
-
-[manaria_compare]: https://slowpics.org/comparison/61e39e1e-d074-4d83-b7c9-b0f4e1861855
-
 
-## Credits and other native 1080p elements
+clip_y = vsutil.get_y(clip)
+descaled = core.descale.Debicubic(clip_y, 1550, 872, b=1/3, c=1/3)
+upscaled = core.resize.Spline36(descaled, 1920, 1080, filter_param_a=1/3, filter_param_b=1/3)
+credit_mask = core.std.Expr([clip_y, upscaled], 'x y - abs')
+credit_mask = kgf.iterate(credit_mask, core.std.Maximum, 4)
+credit_mask = kgf.iterate(credit_mask, core.std.Inflate, 2)
+credit_mask = credit_mask.std.Binarize(0.05)
 
-There is one very, very important thing
-to keep in mind when descaling:
+rescaled = nnedi3_resample(descaled, 1920, 1080)
```

Comment:
`nnedi3_resample` is undefined in this script.

Reply:
Technically resolved this issue but there's another undefined thing, I'll make a separate suggestion.